src/modules/fancyzones/FancyZonesLib/MonitorUtils.cpp

```diff
@@ -359,8 +359,9 @@ namespace MonitorUtils
                 destMi.cbSize = sizeof(destMi);
                 if (GetMonitorInfo(monitor, &destMi))
                 {
-                    RECT newPosition = FitOnScreen(placement.rcNormalPosition, originMi.rcWork, destMi.rcWork);
-                    FancyZonesWindowUtils::SizeWindowToRect(window, newPosition);
+                    RECT newSize = placement.showCmd == SW_SHOWMAXIMIZED ? destMi.rcWork : placement.rcNormalPosition;
+                    RECT newPosition = FitOnScreen(newSize, originMi.rcWork, destMi.rcWork);
```

Comment:
I don't see the need to change the window size. With this code, when you restore the window from a maximized state, it keeps its size equal to the work area size. But if we use just `rcNormalPosition,` it will return to the previous size when restored.
So I think this change should be reverted.

Reply:
yes you are right, I have updated. This logic is for smooth UI animation, avoid flash window when quick change size to maximized, but it better to keep `rcNormalPosition`.